### ISSUE TITLE
Implement LIST command for files in a directory

### ISSUE THREAD

**Issue by gmaclennan** [appium/node-adb-client#14]
ADB sync services supports a [LIST](https://android.googlesource.com/platform/system/core/+/master/adb/SYNC.TXT#35) command for listing files in a directory.

**Comment by cstyan** [comment]
Hey @gmaclennan , can you give an example of how to use the `LIST` command? `adb sync <remote dir> -l` doesn't work for me.

Also, is there any noticeable difference in the output of the `LIST` command and just running `shell ls <remote dir>`?

**Comment by gmaclennan** [comment]
I'm not sure the `LIST` synchronous protocol request maps to the command line option. `adb sync` refers to synchronizing a local and remote directory, and the `-l` option I think means list what will be copied. `sync` within the protocol I understand refers to synchronous requests: `LIST`, `RECV`, `SEND`, `STAT`. 

`LIST`, as I understand it from [the docs](https://android.googlesource.com/platform/system/core/+/master/adb/SYNC.TXT#35), lists files in a directory, and should be similar output to `shell ls <remote dir>`. However, the output `shell ls` is hard to parse, and is liable to bugs: http://mywiki.wooledge.org/ParsingLs

I don't know what the `LIST` command will output other than what is specified in the docs. I was interested in it because we want to copy all files within a known folder, e.g. `/sdcard/myfolder`, so I would want to list the contents first and then pull each one. The `file mode` returned by `LIST` would be useful for identifying directories and recursively finding files.

**Comment by cstyan** [comment]
Alright, so `LIST` over `ls` would be useful. I'll put this on my TODO list. Unfortunately until I figure out how to run the command I can't start on this.

SYNC as far as I can tell is just a subset of the regular ADB protocol, for example sync is used for all the stream related functions. `adb push` actually sends `sync:` and then `send` from the SYNC documentation. 

https://github.com/appium/node-adb-client/blob/master/lib/adb-device.js#L192-L214
